TransactionsLib.js

```diff
@@ -0,0 +1,93 @@
+const moment = require('moment');
+const loggerMessages = require('./loggerMessages');
+const moneyConverter = require('./moneyConverter');
+const fs = require('fs');
+
+class Transaction {
+    constructor(transactionDetails, logger) {
+
+        if (transactionDetails.length !== 5) {
+            loggerMessages.logFatal('Wrong number of parameters for Transaction constructor\n' +
+                                `(have ${transactionDetails.length} instead of 5)`, logger)
+        }
+
+        const [dateString, from, to, narrative, amountStrig] = transactionDetails;
+
+        this.date = parseTime(dateString, logger);
+        this.from = from;
+        this.to = to;
+        this.narrative = narrative;
+        this.ammount = moneyConverter.convertMoneyStringToInt(amountStrig, logger);
+    
+    }
+
+}
+
+function parseTime(dateString, logger) {
+    
+    // CVS format
+    date = moment(dateString, 'DD/MM/YYYY');
+
+    if (!date.isValid()) {
+
+        // If no. days since 1900-01-01 (XML format)
+        if (Number.isInteger(+dateString)) {
+
+            const baseDate = moment('19000101');
+            return date = baseDate.add((+dateString), 'days');

```

Comment:
Why are you returning + assigning in one line instead of just `return baseData.add(...)`?